### DISCUSSION TITLE
Is there a Draw Plugin?

### DISCUSSION THREAD

**Discussion by KushalTalkingLands** [maplibre/maplibre-react-native#570] [Q&A] [answered]
I am trying to use maplibre-gl-draw in react native but it is giving error mapRef.current.addControl is not a function (it is undefined ). How to solve it ?

**Comment by KiwiKilian** [comment, marked as answer]
Not sure which package you are exactly trying to use, as there is mostly `mapbox-gl-draw`. But as the name states it's for Mapbox/MapLibre GL JS, so it works only in the browser, not MapLibre React Native. AFAIK there is not MapLibre Native plugin or anything similar. You will have to build this yourself.

**Reply by KushalTalkingLands** [reply]
Is mapbox-gl-draw compatible ?

**Reply by KiwiKilian** [reply]
No, `mapbox-gl-draw` is not compatible with MapLibre React Native (this repository). It's only compatible with `maplibre-gl-js`, which is MapLibre for the Browser, not native apps/React Native.
